R/PipeOpBreslow.R

```diff
@@ -0,0 +1,181 @@
+#'
+#' @section Dictionary:
+#' This [PipeOp][mlr3pipelines::PipeOp] can be instantiated via the
+#' [Dictionary][mlr3misc::Dictionary] [mlr_pipeops][mlr3pipelines::mlr_pipeops]
+#' or with the associated sugar function [po()][mlr3pipelines::po]:
+#' ```
+#' PipeOpBreslow$new(learner)
+#' mlr_pipeops$get("breslowcompose", learner)
+#' po("breslowcompose", learner, overwrite = TRUE)
+#' ```
+#'
+#' @section Input and Output Channels:
+#' [PipeOpBreslow] is like a [LearnerSurv].
+#' It has one input channel, named `input` that takes a [TaskSurv] during training
+#' and another [TaskSurv] during prediction.
+#' [PipeOpBreslow] has one output channel named `output`, producing `NULL` during
+#' training and a [PredictionSurv] during prediction.
+#'
+#' @section State:
+#' The `$state` slot stores the `times` and `status` survival target variables of
+#' the train [TaskSurv] as well as the `lp` predictions on the train set.
+#'
+#' @section Parameters:
+#' The parameters are:
+#' * `overwrite` :: `logical(1)` \cr
+#'    If `FALSE` (default) then the compositor does nothing and returns the
+#'    input `learner`'s [PredictionSurv].
+#'    If `TRUE` or in the case that the input `learner` doesn't have `distr`
+#'    predictions, then the `distr` is overwritten with the `distr` composed
+#'    from `lp` and the train set information using [breslow].
+#'    This is useful for changing the prediction `distr` from one model form to
+#'    another.
+#' @seealso [pipeline_distrcompositor]
+#' @export
+#' @family survival compositors
+#' @examples
+#' \dontrun{
+#' if (requireNamespace("mlr3pipelines", quietly = TRUE)) {
+#'   library(mlr3)
+#'   library(mlr3pipelines)
+#'   task = tsk("rats")
+#'   part = partition(task, ratio = 0.8)
+#'   train_task = task$clone()$filter(part$train)
+#'   test_task  = task$clone()$filter(part$test)
+#'
+#'   learner = lrn("surv.coxph") # learner with lp predictions
+#'   b = po("breslowcompose", learner = learner, overwrite = TRUE)
+#'
+#'   b$train(list(train_task))
+#'   p = b$predict(list(test_task))[[1L]]
+#' }
+#' }
+PipeOpBreslow = R6Class("PipeOpBreslow",
+  inherit = mlr3pipelines::PipeOp,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    #' @param learner ([LearnerSurv])\cr
+    #' Survival learner which must provide `lp`-type predictions
+    #' @param id (character(1))\cr
+    #' Identifier of the resulting object.
+    initialize = function(learner, id = "po_breslow", param_vals = list(overwrite = FALSE)) {
+      if ("lp" %nin% learner$predict_types) {
+        stopf("Learner %s must provide lp predictions", learner$id)
+      }
+
+      private$.learner = as_learner(learner, clone = TRUE)
+
+      # PipeOp-specific ParamSet
+      ps_pipeop = ps(
+        overwrite = p_lgl(default = FALSE, tags = c("predict"))
+      )
+      # add learner's ParamSet
+      param_set = ParamSetCollection$new(
+        list(private$.learner$param_set, ps_pipeop)
+      )
```

Comment:
So, now this creates the following small issue:
![image](https://github.com/mlr-org/mlr3proba/assets/10909555/5fb36999-a0e8-4082-b8f1-ca66f8fee935)

Reply:
How to have `learner_id.xyz` for the parameters and not `po_breslow.xyz` (only overwrite is in that category)?

Reply:
work-around using same trick as in `PipeOpLearnerCV`